pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -193,6 +194,11 @@ func extractLibInfo(pod *corev1.Pod, containerRegistry string) []libInfo {
 		libInfoList = extractLibrariesFromAnnotations(pod, containerRegistry, libInfoMap)
 	}
 
+	// Try getting the languages from the owner of the pod from workloadmeta
+	if len(libInfoList) == 0 {
+		libInfoList = extractLibrariesFromOwnerAnnotations(pod, containerRegistry)
```

Comment:
I think we should document this behavior somewhere: if annotations are added manually, automatic language detection is not enabled.

Reply:
Can we have it behind a feature flag as well ? In case we need to disable it and annotations injections are already there

Reply:
I think it's also worth adding a unit test for this case.

Reply:
Imo this `if  len(libInfoList) == 0` 
should be part of the `if shouldInject(pod) {` (line 192).  because we want to inject lib only if they enable it by default or for a specific namespace.